### ISSUE TITLE
Snakemake 7 does not create SLURM dependencies when --immediate-submit flag is used

### ISSUE THREAD

**Issue by sreichl** [snakemake/snakemake#1715]
First of all: Thanks for Snakemake! I love it and it makes my life so much easier!!!

**Snakemake version**
Snakemake 7.

**Describe the bug**
I have been using Snakemake with an adapted SLURM cluster profile with the --immediate-submit flag, so Snakemake submitted always all the required jobs with the respective dependencies at once.

With Snakemake 7 the profile and job submission still work fine but **no dependencies are provided** hence all jobs that depend on running jobs fail/don't start immediately. Repeated iteratively until no unfinished jobs are left works as well, but is of course no solution.

**Additional context**
SLURM profile I am currently using: https://github.com/epigen/unsupervised_analysis/tree/main/config/slurm.cemm
original SLURM profile cookiecutter: https://github.com/Snakemake-Profiles/slurm 

If you have a solution, an idea, or know someone who is working with Snakemake and/or SLURM in more depth, please let me know.

**Comment by SichongP** [comment]
I'm having the same problem. Do you know what was the last version that worked? That might help pin point the cause.

**Comment by sreichl** [comment]
I do not know the exact last version but I know for sure that it works with 6.0.3. 

Hope this helps! Please let me know if you figure out how to make it work. 

Thanks & Cheers,
Stephan

**Comment by Hocnonsense** [comment]
@SichongP
> I'm having the same problem. Do you know what was the last version that worked? That might help pin point the cause.

My analyze may help https://github.com/snakemake/snakemake/issues/1865
(maybe I should close that issue and move it here)

**Comment by sreichl** [comment]
Hi @Hocnonsense,
thanks for your analysis.

Did you get it to work? Does your PR fix the issue?

Thanks & Cheers,
Stephan

**Comment by Hocnonsense** [comment]
Hi, a easy way to fix it, is add some line to https://github.com/snakemake/snakemake/blob/dc45ccb9ee94e8fcdf386c8598bbae57e319ba10/snakemake/executors/__init__.py#L1272-L1281 like this:
https://github.com/snakemake/snakemake/pull/1851/commits/6649305b88d4bdc38a226e4d178f845836ff37ed#diff-438f3317205fd7130727d0589d2fc1a6c2e1f6fc48c2c04d354a8a09b91ba2f4

This could quickly fix this problem. However, after I review the code and noticed PR 41a5071, I revert this fix as it make thing more complex.

I'm still puzzled about the purpose of this PR, so I have no new PR now. 

Do you have any idea? Thanks!

**Comment by sreichl** [comment]
Hi,
I do not have any new ideas as of now, sorry.

Cheers,
Stephan

**Comment by johanneskoester** [comment]
Hi folks. I've just merged PR #1851. If that fixes your issue, please close it. If not, feel free to keep it open. Note that I might forget to check this routinely, so feel free to ping me via discord.

**Comment by sreichl** [comment]
Hi @johanneskoester,
I installed the latest Snakemake version (7.15.2.) and it worked!

Thanks a lot, @Hocnonsense for the PR, you made my day!

Cheers,
Stephan
